test/langtools/tools/javac/api/TestJavacTaskWithWarning.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug     8348212
+ * @summary Ensure the warn() phase executes when the compiler is invoked via the API
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ * @compile TestJavacTask.java TestJavacTaskWithWarning.java
+ * @run main TestJavacTaskWithWarning
+ */
+
+import com.sun.tools.javac.api.JavacTaskImpl;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+import javax.tools.JavaFileObject;
+import javax.tools.ToolProvider;
+
+public class TestJavacTaskWithWarning extends TestJavacTask {
```

Comment:
We don't seem to have any other test doing this... what are you using of the super class?

Reply:
probably just add
```
static final JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
static final StandardJavaFileManager fm = compiler.getStandardFileManager(null, null, null);
 ```
 to the test and get rid of the dependency? Seems to be what other tests are doing in here...

Reply:
> probably just add...

Agreed, that's simpler. At first when writing this test I thought I was going to utilize more stuff from the superclass (`TestJavacTask`) but in the end didn't need it.

Cleaned up in a9b5f2d63cd.